File: solutions/go/scrabble-score/1/scrabble_score.go
package scrabble

import "strings"

func Score(word string) (points int) {
	for _, char := range strings.ToLower(word) {
		if strings.ContainsRune("aeioulnrst", char) {
			points++
		} else if strings.ContainsRune("dg", char) {
			points += 2
		} else if strings.ContainsRune("bcmp", char) {
			points += 3
		} else if strings.ContainsRune("fhvwy", char) {
			points += 4
		} else if char == 'k' {
			points += 5
		} else if strings.ContainsRune("jx", char) {
			points += 8
		} else if strings.ContainsRune("qz", char) {
			points += 10
		}
	}

	return
}

File: solutions/ruby/scrabble-score/1/scrabble_score.rb
# frozen_string_literal: true

# Scrabble score
class Scrabble
  def initialize(word)
    @word = word
  end

  def score
    @word.downcase.chars.reduce(0) { |acc, char| acc + char_score(char) }
  end

  private

  def char_score(char)
    if 'aeioulnrst'.include?(char) then 1
    elsif 'dg'.include?(char) then 2
    elsif 'bcmp'.include?(char) then 3
    elsif 'fhvwy'.include?(char) then 4
    elsif char == 'k' then 5
    elsif 'jx'.include?(char) then 8
    elsif 'qz'.include?(char) then 10
    else
      0
    end
  end
end
